annotation_store/store.py

```diff
@@ -153,27 +144,30 @@ def _verify_user(self, user_id, raise_exception=True):
         return result
 
     def _get_tool_provider(self):
-        params = self.request.LTI['launch_params']
-        tool_provider = DjangoToolProvider(CONSUMER_KEY, LTI_SECRET, params)
-        return tool_provider
+        if 'launch_params' in self.request.LTI:
+            params = self.request.LTI['launch_params']
+            return DjangoToolProvider(CONSUMER_KEY, LTI_SECRET, params)
+        return DjangoToolProvider(CONSUMER_KEY, LTI_SECRET)
 
-    def _lti_grade_passback(self, is_graded=False, status_code=None, result_score=1):
-        self.logger.debug("LTI Grade Passback: is_graded=%s status_code=%s result_score=%s" % (is_graded, status_code, result_score))
-        if not is_graded:
-            return
-        if status_code != 200:
-            self.logger.info("LTI Grade Passback aborted because status_code=%s" % status_code)
+
+    def lti_grade_passback(self, score=1.0):
+        tool_provider = self._get_tool_provider()
+        if not tool_provider.is_outcome_service():
+            self.logger.debug("LTI consumer does not expect a grade for the current user and assignment")
             return
+        self.logger.info("Initiating LTI Grade Passback: score=%s" % score)
         try:
-            outcome = self._get_tool_provider().post_replace_result(result_score)
+            outcome = tool_provider.post_replace_result(score)
+            self.logger.info(vars(outcome))
             if outcome.is_success():
                 self.logger.info(u"LTI grade request was successful. Description: %s" % outcome.description)
             else:
                 self.logger.error(u"LTI grade request failed. Description: %s" % outcome.description)
-            self.grade_passback_outcome = outcome
+            print outcome
```

Comment:
This print statement should be removed and/or replaced with a call to the logger's `debug()` method. Probably can be removed since there's a log message for both the success and failure cases that precede it.

Reply:
sorry! missed that one! Thank you